batch-jberet/src/main/java/org/wildfly/extension/batch/jberet/deployment/BatchDeploymentDescriptorParser_1_0.java

```diff
@@ -61,6 +62,8 @@ public BatchEnvironmentMetaData parse(final XMLExtendedStreamReader reader, fina
 
             // Process the job repository
             if (element == Element.JOB_REPOSITORY) {
+                executionRecordsLimit = parseExecutionRecordsLimit(reader);
```

Comment:
What is the reason for changing parsing in the 1.0 parser?

Reply:
I made this change in order to make the `BatchDeploymentDescriptorParser_1_0` more extendable, to simplify the new `BatchDeploymentDescriptorParser_3_0` class. I would have to copy paste the whole `parse()` method otherwise. This way I just need to override the `parseExecutionRecordsLimit()` method in version 3 parser, and the rest of the code still lives in the original classes only. In version 1 the method doesn't accept any attributes and just returns `null`, so it doesn't add any new features to version 1.

The same approach was used in version 2 with `parseJdbcJobRepository()` method, so it seemed like a good idea to follow the approach.

Reply:
Sorry yes I see you throw an error if called - please disregard my comment.